addons/l10n_dk_nemhandel/wizard/nemhandel_registration.py

```diff
@@ -0,0 +1,319 @@
+from odoo.addons.l10n_dk_nemhandel.tools.demo_utils import handle_demo
+
+
+class NemhandelRegistration(models.TransientModel):
+    _name = 'nemhandel.registration'
+    _description = "Nemhandel Registration"
+
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        required=True,
+        default=lambda self: self.env.company,
+    )
+    contact_email = fields.Char(
+        related='company_id.nemhandel_contact_email', readonly=False,
+        required=True,
+    )
+    edi_mode = fields.Selection(
+        string='EDI mode',
+        selection=[('demo', 'Demo'), ('test', 'Test'), ('prod', 'Live')],
+        compute='_compute_edi_mode',
+        inverse='_inverse_edi_mode',
+        readonly=False,
+    )
+    edi_user_id = fields.Many2one(
+        comodel_name='account_edi_proxy_client.user',
+        string='EDI user',
+        compute='_compute_edi_user_id',
+    )
+    edi_mode_constraint = fields.Selection(
+        selection=[('demo', 'Demo'), ('test', 'Test'), ('prod', 'Live')],
+        compute='_compute_edi_mode_constraint',
+        help="Using the config params, this field specifies which edi modes may be selected from the UI"
+    )
+    phone_number = fields.Char(related='company_id.nemhandel_phone_number', readonly=False)
+    l10n_dk_nemhandel_proxy_state = fields.Selection(related='company_id.l10n_dk_nemhandel_proxy_state', readonly=False)
+    verification_code = fields.Char(related='edi_user_id.nemhandel_verification_code', readonly=False)
+    identifier_type = fields.Selection(related='company_id.nemhandel_identifier_type', readonly=False, required=True)
+    identifier_value = fields.Char(related='company_id.nemhandel_identifier_value', readonly=False, required=True)
+    receiver_registration = fields.Boolean(string='Register as a receiver', default=True)
+
+    # -------------------------------------------------------------------------
+    # ONCHANGE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.onchange('identifier_value')
+    def _onchange_identifier_value(self):
+        for wizard in self:
+            if wizard.identifier_value:
+                wizard.identifier_value = ''.join(char for char in wizard.identifier_value if char.isalnum())
+
+    @api.onchange('phone_number')
+    def _onchange_phone_number(self):
+        for wizard in self:
+            if wizard.phone_number:
+                wizard.company_id._sanitize_nemhandel_phone_number(wizard.phone_number)
+                with contextlib.suppress(phonenumbers.NumberParseException):
+                    parsed_phone_number = phonenumbers.parse(
+                        wizard.phone_number,
+                        region=self.company_id.country_code,
+                    )
+                    wizard.phone_number = phonenumbers.format_number(
+                        parsed_phone_number,
+                        phonenumbers.PhoneNumberFormat.E164,
+                    )
```

Comment:
Could we use compute instead of onchange for maintainability onchange is pain ahah

Reply:
I don't see how. A compute that would change its own value seems weird (and a compute would proc too fast).